### PR TITLE
Adds sub-interface support for nxos_interface and nxos_ip_interface

### LINKED ISSUE #29164: Cannot create subinterfaces using nxos_interface
Cannot create sub-interface using nxos_interface

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
nxos_interface

##### ANSIBLE VERSION
```
ansible 2.4.0.0
  config file = /home/johannlo/ansible/core/ansible.cfg
  configured module search path = [u'/home/johannlo/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/johannlo/venv-ansible2.4/local/lib/python2.7/site-packages/ansible
  executable location = /home/johannlo/venv-ansible2.4/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### CONFIGURATION
```
[defaults]
hostfile = hosts
host_key_checking = False
```

##### OS / ENVIRONMENT
nxos.7.0.3.I7.1.bin
Ubuntu 16.04 LTS

##### SUMMARY
using nxos_interface or nxos_ip_interface, you cannot create a subinterface. The parser appears to drop the dot (see attached debug). 

Also, there appears to be NO WAY using standard nxos modules to add the dot1q encapsulation. I have resorted to doing all this manually using nxos_config as a workaround.

##### STEPS TO REPRODUCE
Play
```
  - name: set physical interfaces
    nxos_interface:
      interface: "{{ item.interface }}"
      admin_state: up
      mode: "{{ item.mode }}"
      provider: "{{ provider }}"
    with_items: "{{ physical_interfaces | default([])  }}"
```

Vars
```
physical_interfaces:
  - interface: Ethernet1/3.80
    mode: layer3
```

##### EXPECTED RESULTS
Create subinterface Ethernet1/3.80
Make it a routed interface and no shut

Also, there appears to be NO WAY using standard nxos modules to add the dot1q encapsulation. I have resorted to doing all this manually using nxos_config as a workaround.

##### ACTUAL RESULTS
See the "interface" keypair - the dot is dropped! 
```
failed: [dc1-core-01] (item={u'interface': u'Ethernet1/3.80', u'mode': u'layer3'}) => {
    "changed": false, 
    "failed": true, 
    "interface": "Ethernet1/380", 
    "invocation": {
        "module_args": {
            "admin_state": "up", 
            "description": null, 
            "fabric_forwarding_anycast_gateway": null, 
            "host": "dc1-core-01", 
            "interface": "Ethernet1/3.80", 
            "interface_type": null, 
            "ip_forward": null, 
            "mode": "layer3", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": 80, 
            "provider": {
                "host": "dc1-core-01", 
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "port": 80, 
                "ssh_keyfile": null, 
                "timeout": 10, 
                "transport": "nxapi", 
                "use_ssl": false, 
                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "validate_certs": false
            }, 
            "ssh_keyfile": null, 
            "state": "present", 
            "timeout": 10, 
            "transport": "nxapi", 
            "url_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "url_username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "use_ssl": false, 
            "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "validate_certs": false
        }
    }, 
    "item": {
        "interface": "Ethernet1/3.80", 
        "mode": "layer3"
    }, 
    "msg": "Invalid Ethernet interface provided."
}
```

### PR DESCRIPTION
**Pull request by kedarkekan**
##### SUMMARY
Existing modules didn't have sub-interface configuration support.

- Adds sub-interface support for nxos_interface and nxos_ip_interface
 - Support dot1q encapsulation on routed sub-interface

Fixes #29164 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
modules/network/nxos

##### ANSIBLE VERSION
```
devel(2.5.0)
```

### CODE REVIEW

File: lib/ansible/modules/network/nxos/nxos_ip_interface.py [hunk 1]
```diff
@@ -155,9 +171,9 @@
 try:
     import ipaddress
 
-    HAS_IPADDRESS = True
+    IPADDRESS_MODULE = True
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_ip_interface.py]: Any reason you're changing this in particular? Most (all?) of the "Is this module installed" markers are named `HAS_[modulename]` for consistency

File: lib/ansible/modules/network/nxos/nxos_ip_interface.py [hunk 2]
```diff
@@ -105,6 +111,16 @@
     addr: 20.20.20.20
     mask: 24
 
+- name: Ensure ipv4 address is configured with tag
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_ip_interface.py]: This seems to be the description of the example above, and vice versa

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/nxos/nxos_ip_interface.py (hunk 1 above)
Comment (kedarkekan) [reply lib/ansible/modules/network/nxos/nxos_ip_interface.py]: I see; thought to have module word explicitly. HAS_SOMETHING does didn't reflect that. Anyways will revert in the interest of consistency.

File: lib/ansible/modules/network/nxos/nxos_ip_interface.py (hunk 2 above)
Comment (kedarkekan) [reply lib/ansible/modules/network/nxos/nxos_ip_interface.py]: Modified

**Review by ganeshrn — APPROVED** [review]

**Review by Qalthos — APPROVED** [review]

**Pull request merged**